backend/src/data_provider/Extractor.py

```diff
@@ -1,16 +1,75 @@
+        transfers_df = self.create_transfers_trips_df(transfers_df, route_ids_df)
+        current_stop_times_df = self.create_stop_times_trips_df_for_service_id(stop_times_df, route_ids_df)
+        day_to_services_dict = self.get_day_to_services_dict(calendar_df)
+        routes_df = self.create_routes_trips_df(trips_df, routes_df, route_ids_df)
+        services_list = self.get_services_list(calendar_df)
+        routes_to_stops_dict = self.create_route_to_stops_dict(stop_times_df, route_ids_df)
+        exception_days = self.create_exception_days_dict(calendar_dates_df)
+        period_df = self.create_period_df(stop_times_df, route_ids_df, routes_df)
+
+        # Floyd extraction
+        current_stop_times_df['service'] = current_stop_times_df.index.get_level_values('service_id')
+        current_stop_times_df = current_stop_times_df.reset_index('stop_sequence')
+        graph = self.extract_graph(stops_df, transfers_df, period_df)
+        extended_graph = self.extend_graph(graph, stops_df)
+        adjacent_stops = self.get_adjacent_stops_dict(stops_df)
+        floyd_transfers_df = self.create_floyd_transfers_df(extended_graph, adjacent_stops)
+        stops_df = self.extend_stops_df(transfers_df, adjacent_stops, stops_df)
+        floyd_graph = self.create_floyd_graph(floyd_transfers_df, stops_df)
+        kernelized_floyd_graph = self.create_kernelized_floyd_graph(floyd_graph, stops_df)
+        distances = self.get_distances(floyd_graph)
+
+        previous_stop_times_df = copy.deepcopy(current_stop_times_df)
+        previous_stop_times_df['departure_time'] = previous_stop_times_df['departure_time'].apply(
+            lambda t: t - 24 * 3600)
+        previous_stop_times_df = previous_stop_times_df[previous_stop_times_df['departure_time'] >= 0]
+
+        next_stop_times_df = copy.deepcopy(current_stop_times_df)
+        next_stop_times_df['departure_time'] = next_stop_times_df['departure_time'].apply(lambda t: t + 24 * 3600)
+        next_stop_times_df = next_stop_times_df[next_stop_times_df['departure_time'] <= 36 * 3600]
+
+        current_stop_times_dict = self.transform_stop_times_df_to_dict(stops_df, current_stop_times_df,
+                                                                            services_list)
+        previous_stop_times_dict = self.transform_stop_times_df_to_dict(stops_df, previous_stop_times_df,
+                                                                             services_list)
+        next_stop_times_dict = self.transform_stop_times_df_to_dict(stops_df, next_stop_times_df, services_list)
+
+        return ExtractedData(floyd_graph, kernelized_floyd_graph, distances, day_to_services_dict,
+                             current_stop_times_dict, previous_stop_times_dict, next_stop_times_dict,
+                             routes_to_stops_dict, adjacent_stops, exception_days, stops_df, routes_df,
+                             stops_df_by_name, stop_times_df)
```

Comment:
This looks very good. BTW can we use the following syntax? This will ensure that arguments are passed correctly without the need manually check their order:
```py
return ExtractedData(floyd_graph=floyd_graph, 
                     kernelized_floyd_graph=kernelized_floyd_graph,
                     ...
```